assets/src/edit-story/components/library/panes/pageTemplates/savedTemplates.js

```diff
@@ -19,27 +19,99 @@
  */
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { useEffect, useState, useCallback, useRef } from 'react';
 
 /**
  * Internal dependencies
  */
+import { useAPI } from '../../../../app/api';
+import { UnitsProvider } from '../../../../units';
+import useRovingTabIndex from '../../../../utils/useRovingTabIndex';
+import PageTemplate from './pageTemplate';
 import TemplateSave from './templateSave';
+import ConfirmPageTemplateDialog from './confirmPageTemplateDialog';
+import useTemplateActions from './useTemplateActions';
 
 const Wrapper = styled.div`
-  position: relative;
-  margin-left: 1em;
+  padding-left: 1em;
+  padding-top: 5px;
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  grid-gap: 8px;
+  overflow-y: scroll;
 `;
 
-function SavedTemplates({ pageSize }) {
+function SavedTemplates({ pageSize, setShowDefaultTemplates }) {
+  const {
+    actions: { getCustomPageTemplates },
+  } = useAPI();
+
+  const [pageTemplates, setPageTemplates] = useState(null);
+  const ref = useRef();
+
+  const loadTemplates = useCallback(() => {
+    getCustomPageTemplates().then(setPageTemplates);
+  }, [getCustomPageTemplates]);
+
+  useEffect(() => {
+    if (!pageTemplates) {
+      loadTemplates();
+    }
+  }, [loadTemplates, pageTemplates]);
+
+  const {
+    isConfirming,
+    handleCloseDialog,
+    handleConfirmDialog,
+    handleKeyboardPageClick,
+    handlePageClick,
+  } = useTemplateActions();
+
+  useRovingTabIndex({ ref });
+
   return (
-    <Wrapper>
-      <TemplateSave pageSize={pageSize} />
-    </Wrapper>
+    <UnitsProvider
```

Comment:
This part will then likely be replaced with a virtual list instead if pagination would cause performance issues here.